### PR TITLE
Adds new black market pin

### PR DESCRIPTION
**Pull request by Jacquerel**
## About The Pull Request

image

Adds a fetching red pin which you can wear to visibly align yourself with the enemies of Nanotrasen, purchaseable from the Black Market Uplink.
This pin's contained RFID chip will automatically cause you to appear on Sec HUDs as a wanted criminal and will aggro securitrons, to prove your dedication to your ideals.

If your convictions are a little less firm, you can also silently pin this onto _other_ people's clothing if they stand still next to you for five seconds... though they might notice that they're suddenly wearing a red badge.

I didn't want this to be a subtype of medal so I made the "you can put this accessory onto someone else" behaviour into a component to sidestep object inheritance.
This has been applied to the detective's spy camera, because it makes it much easier to turn someone into a mole.

This also adds a trait which makes security hate you which I guess someone could use in a novelty bar drink at some point or something.

## Why It's Good For The Game

It's funny
I think we need more neat things in the black market to make it sometimes worth using

## Changelog

:cl:
add: Added subversive pins to the black market uplink which make security hate you
add: The detective's spy cam can now be conveniently pinned onto people in the same manner as medals
/:cl:

### CODE REVIEW

File: code/_globalvars/traits/_traits.dm [hunk 1]
```diff
@@ -153,6 +153,7 @@ GLOBAL_LIST_INIT(traits_by_type, list(
 		"TRAIT_CLUMSY" = TRAIT_CLUMSY,
 		"TRAIT_COAGULATING" = TRAIT_COAGULATING,
 		"TRAIT_CORPSELOCKED" = TRAIT_CORPSELOCKED,
+		"TRAIT_CRIMINAL_SYMPATHISER" = TRAIT_CRIMINAL_SYMPATHISER,
```
Comment (Mothblocks) [inline code/_globalvars/traits/_traits.dm:156]: abc

Comment (Jacquerel) [reply code/_globalvars/traits/_traits.dm:156]: what am i missing isnt this already in the correct alphabetical location?

File: code/datums/components/pinnable_accessory.dm [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+		return COMPONENT_INCOMPATIBLE
+	src.silent = silent
+	src.pinning_time = pinning_time
+	src.on_pre_pin = on_pre_pin
+
+/datum/component/pinnable_accessory/RegisterWithParent()
+	RegisterSignal(parent, COMSIG_ITEM_PRE_ATTACK, PROC_REF(on_pre_attack))
+
+/datum/component/pinnable_accessory/UnregisterFromParent()
+	UnregisterSignal(parent, COMSIG_ITEM_PRE_ATTACK)
```
Comment (MrMelbert) [inline code/datums/components/pinnable_accessory.dm:22]: Suggestion: Try using the new `COMSIG_ITEM_INTERACTING_WITH_ATOM` signal.

File: code/_globalvars/traits/_traits.dm (hunk 1 above)
Comment (Mothblocks) [reply code/_globalvars/traits/_traits.dm:156]: my eyes deceived me

File: code/modules/clothing/under/accessories/badges.dm [hunk 3]
```diff
@@ -203,3 +203,36 @@ GLOBAL_LIST_INIT(pride_pin_reskins, list(
...
+	. = ..()
+	AddComponent(/datum/component/pinnable_accessory, silent = TRUE, pinning_time = 5 SECONDS)
+
+/obj/item/clothing/accessory/anti_sec_pin/attach(obj/item/clothing/under/attach_to, mob/living/attacher)
+	. = ..()
+	if (!.)
+		return FALSE
+
+	var/target = ishuman(attach_to.loc) ? attach_to.loc : attach_to
+	log_combat(attacher, target, "pinned a suspicious pin onto", src)
```
Comment (Ghommie) [inline code/modules/clothing/under/accessories/badges.dm:223]: This should be a bit more specific. Not many will know about it anyway.

File: code/datums/components/pinnable_accessory.dm [hunk 4]
```diff
@@ -0,0 +1,80 @@
+/// This accessory can be pinned onto someone else
+/datum/component/pinnable_accessory
```
Comment (JohnFulpWillard) [inline code/datums/components/pinnable_accessory.dm:2]: this would be swag AF with detective's spy kit

Comment (Ghommie) [reply code/datums/components/pinnable_accessory.dm:2]: 👆 This!

**Review by Wallemations — APPROVED** [review]
cute sprite

File: code/__DEFINES/traits/declarations.dm [hunk 5]
```diff
@@ -1011,4 +1011,7 @@ Remember to update _globalvars/traits.dm if you're adding/removing/renaming trai
 ///Trait given when a mob has been tipped
 #define TRAIT_MOB_TIPPED "mob_tipped"
 
+/// Trait which self-identifies as an enemy of the law
+#define TRAIT_CRIMINAL_SYMPATHISER "criminal_sympathiser"
```
Comment (Fikou) [inline code/__DEFINES/traits/declarations.dm:1015]:
```suggestion
#define TRAIT_ALWAYS_WANTED "always_wanted"
```

**Comment by radarseas** [comment]
It would be funny to have chamelion pins that do the same thing for tots... sneakily replace someone's mime/clown pin or allergies dogtag with one of these and watch beepsky stun and cuff them.

**Pull request merged**